modules/nw-ingress-integrating-route-secret-certificate.adoc

```diff
@@ -0,0 +1,71 @@
+//
+// * ingress/routes.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ingress-integrating-route-secret-certificate_{context}"]
+= Secure route with external certificates in TLS secrets
+
+:FeatureName: Secure route with external certificates in TLS secrets
+include::snippets/technology-preview.adoc[]
+
+The following workflow describes how to integrate third-party certificate management solutions, using the certificate reference field outlined under API Extensions.
+
+A new `.spec.tls.externalCertificate` field has been added to the route's `.spec.tls` configuration. This field can be used to specify the name of a secret containing the certificate data.
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the cluster-admin role.
+
+* You must have generated the serving certificate using third-party systems.
+
+* Make sure the `secret` is categorized as `kubernetes.io/tls`, and include both `tls.key` and `tls.crt` keys within the data or `stringData` field of the secret configuration.
+
+.Procedure
+
+* Create a role in the same namespace as the secret containing the certificate generated.
++
+[source,terminal]
+----
+$ oc create role secret-reader --verb=get,list,watch --resource=secrets --resource-name=<secret-name>
+--namespace=<current-namespace>
+----
+
+* Create a rolebinding in the same namespace as the secret and bind the router service account to the newly created role.
++
+[source,terminal]
+----
+$ oc create rolebinding secret-reader-binding --role=secret-reader --serviceaccount=openshift-ingress:router --namespace=<current-namespace>
+----
+
+* To enable access to a user workload, generate a new Route referencing the generated secret in the .`spec.tls.externalCertificate` field from the previous step.
++
+.Example output
+[source,terminal]
```

Comment:
I think you might want to use `[source,yaml]` here, to format it as yaml file, instead of bash.